### ISSUE TITLE
Unable to destroy account with attached SSO policies

### ISSUE THREAD

**Issue by karmaniverous** [karmaniverous/metastructure#18]
Adding `action: detach` to an account with an attached SSO policy produces the following error:

```text
│ Error: deleting IAM Policy (arn:aws:iam::205930623657:policy/SSOTerraformStateWriter): operation error IAM: DeletePolicy, https response error StatusCode: 409, RequestID: 1d5681bf-eb04-46d4-93f8-c18c60d014ac, DeleteConflict: Cannot delete a policy attached to entities.
```

On examination, the account has been removed from `sso.reference.group_account_permission_set_policies`, so it no longer shows up as a dependency of the related `aws_ssoadmin_customer_managed_policy_attachment` resource, for example here:

```tf
###############################################################################
# Add policy SSOTerraformStateWriter to permission set TerraformAdmin.
###############################################################################
resource "aws_ssoadmin_customer_managed_policy_attachment" "terraform_admin_sso_terraform_state_writer" {
  customer_managed_policy_reference {
    name = "SSOTerraformStateWriter"
  }
  instance_arn       = local.sso_arn
  permission_set_arn = aws_ssoadmin_permission_set.terraform_admin.arn
  depends_on = [
    aws_iam_policy.dev_sso_terraform_state_writer,
    aws_iam_policy.log_archive_sso_terraform_state_writer,
    aws_iam_policy.master_sso_terraform_state_writer,
    aws_iam_policy.prod_sso_terraform_state_writer,
    aws_iam_policy.test_sso_terraform_state_writer,
    aws_iam_policy.shared_services_sso_terraform_state_writer
  ]
}
```

It may be that excluding actioned accounts from `sso.reference.group_account_permission_set_policies` was the wrong choice. Investigating...

**Comment by karmaniverous** [comment]
Bit of a circular dependency here. The attachment resource depends on the policy--this was necessary, or the attachment was created before the policy existed--but detaching the policy is prerequisite to removing it.

I think we need another action here, like `detach`. This action would not destroy the account resources, but WOULD remove the attachment. So you'd run MS once with `detach`, and then again with `destroy`.

Another alternative: `destroy` doesn't remove the resource. That will happen anyway when you remove it from the config.

Investigating...

**Comment by karmaniverous** [comment]
Data points: 

`destroy` is currently not consumed by any template.

Any `action` prevents account creation.

Any `action` excludes an item from output.

Any `action` uses account id from config instead of resource in provider.

So if we only prevent account creation for `action = remove` then the account will still exist until it is removed from config and this error should go away.

Begs the question... is `destroy` the right term? Sill applies to OUs & users without any of these complications, but you could really get the same effect by just deleting the relevant config. `remove` is a different story.

Let's try it out...

**Comment by karmaniverous** [comment]
Ok I was able to remove the policies but then when I deleted the account from config the process failed because there was no provider. Which was why we created the `destroy` action in the first place. 

So it looks like we really do need a `detach` action for accounts. And while we're at it, we do NOT need a `destroy` action for SSO users & OUs, just `remove`. So let's refactor [`actionableSchema`](https://github.com/karmaniverous/metastructure/blob/5e0a0397a5644163181dbb112cc308966efcbc80/src/Config.ts#L10-L13) into multiple schemata to reflect that, say:

* removable: has `id` + `remove` action.
* destroyable: has `id` + `remove`, `detach`, and `destroy` actions. This would only apply to accounts.
